src/persist/src/indexed/metrics.rs

```diff
@@ -55,7 +55,10 @@ impl BlobMetricsByType {
 }
 
 /// Persistence related monitoring metrics.
-#[derive(Clone, Debug)]
+///
+/// Intentionally not Clone because we expect this to be passed around in an
```

Comment:
We could also just have `Metrics::register_with` return an Arc<Metrics> instead of a Metrics. Thoughts?

Reply:
It would certainly get rid of all the new `Arc::new(...)` calls... but I'm afraid I don't have a strong preference.

Side q: it works to just make it an `Arc<Metrics>` instead of sth like `Arc<Mutex<Metrics>>` because the various metrics structs/impls have/use interior mutability and/or are atomics, right?

Reply:
I was thinking in case we have to do any mutations to it before we wrap it in an arc, but I don't have any examples of why we'd need that atm. I'll leave it as-is, we can always change it later.

> Side q: it works to just make it an Arc<Metrics> instead of sth like Arc<Mutex<Metrics>> because the various metrics structs/impls have/use interior mutability and/or are atomics, right?

Yeah they're all atomics